### ISSUE TITLE
Dogfooding Test: Score Parsing Robustness with Real LLM Outputs

### ISSUE THREAD

**Issue by joshreini1** [truera/trulens#2489]
## Description

`test_feedback_score_generation.py` tests the score parser (`re_0_10_rating`) with hardcoded strings. But we never test whether real LLM outputs from our feedback functions parse correctly. If a prompt change causes the LLM to output scores in a slightly different format, we'd miss the regression.

This test uses TruLens feedback functions to generate actual LLM responses and verifies the score parser handles them.

## Proposed Change

Create `tests/dogfooding/test_score_parsing_real.py` that:

1. For several feedback functions (relevance, groundedness, coherence, sentiment), calls the underlying `_create_chat_completion` with the actual system/user prompts and known inputs
2. Captures the raw LLM response string
3. Runs the response through the score parser (`feedback_generated.re_0_10_rating` or JSON extraction logic)
4. Asserts the parser extracts a valid score without raising `ParseError`
5. Asserts the extracted score is in the valid range

This catches regressions where prompt template changes cause LLM output format drift that breaks score extraction.

## Test Infrastructure

- Mark with `@pytest.mark.optional`
- Use `OpenAI(model_engine="gpt-4o-mini")`
- Test at least 5 different feedback functions to cover different prompt/response formats (LIKERT, JSON, numeric)

## Files to Reference

- `src/feedback/trulens/feedback/generated.py` — `re_0_10_rating` and other score extraction logic
- `src/feedback/trulens/feedback/llm_provider.py` — `generate_score()` at line 179, `generate_score_and_reasons()` at line 282
- `tests/unit/test_feedback_score_generation.py` — existing hardcoded score parsing tests

## Running Tests

```bash
TEST_OPTIONAL=true poetry run pytest tests/dogfooding/test_score_parsing_real.py -v -s
```
